src/Workspaces/Core/Portable/Workspace/IsolatedAnalyzerReferenceSet.Core.cs

```diff
@@ -179,27 +249,52 @@ public static async partial ValueTask<ImmutableArray<AnalyzerReference>> CreateI
             if (s_checksumToReferenceSet.TryGetValue(checksum, out var weakIsolatedReferenceSet) &&
                 weakIsolatedReferenceSet.TryGetTarget(out var isolatedAssemblyReferenceSet))
             {
-                return isolatedAssemblyReferenceSet.AnalyzerReferences;
+                return isolatedAssemblyReferenceSet.GetAnalyzerReferences(checksum);
             }
 
-            isolatedAssemblyReferenceSet = new IsolatedAnalyzerReferenceSet(analyzerReferences, assemblyLoaderProvider);
+            // This set of references have not been computed yet.  We have three options:
+            //
+            // 1. These are the very first time we're seeing any references.  Create a fresh isolated set, and add these new
+            //    reference to it.  New references can also be added to this in the future as long as there are no conflicts
+            //    with what's in the set already.
```

Comment:
At first this statement threw me off, since technically adding another analyzer can change the behavior of other analyzers in that same set, due to that back-compat behavior we have where we pick stuff from other folders if somebody forgot to ship their own dependency. For example, if an analyzer depends on v3 of Newtonsoft.Json but forgot to ship it, it might work if a different analyzer shipped v1 and we use that. But hey, you add a new analyzer that ships v2 and something different might happen now.

But I guess we don't care here? Prior to this change there was no isolation between projects -- we effectively had a single loader for the entire process, so there was no guarantee of anything being sane. Nothing actually changes here, right? The intermediate state of this PR where we had one loader per set of analyzers effectively meant we would be more isolated about this edge case....but it's very hard for me to argue that's worth the extra overhead of it.

Reply:
> But I guess we don't care here? 

Yes.  I very much do not care.  The logic we simply have is: we'll assume everything is ok *except* for the case where an actual analyzer-file-ref that we were pointing at actually changes.  In that case, we reload.  Everything else is just undefined.